OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -322,9 +340,9 @@ public SubCell GetAvailableSubCell(Actor self, CPos cell, BlockedByActor check,
 		/// <remarks>This logic is replicated in <see cref="HierarchicalPathFinder.ActorIsBlocking"/> and
 		/// <see cref="HierarchicalPathFinder.ActorCellIsBlocking"/>. If this method is updated please update those as
 		/// well.</remarks>
-		bool IsBlockedBy(Actor actor, Actor otherActor, Actor ignoreActor, bool ignoreSelf, CPos cell, BlockedByActor check, CellFlag cellFlag)
+		bool IsBlockedBy(Actor actor, Actor otherActor, Actor ignoreActor, CPos cell, BlockedByActor check, CellFlag cellFlag)
 		{
-			if (otherActor == ignoreActor || (ignoreSelf && otherActor == actor))
+			if (otherActor == ignoreActor)
```

Comment:
you do not us the function argument now?
shouldn't you test `check`?

Reply:
> you do not us the function argument now?

`ignoreSelf ` is checked by the loop in the callsite instead now, so it is no longer needed here.

> shouldn't you test check?

The `check` argument is used within the `IsBlockedBy` function. I haven't changed any logic regarding `check` in this PR.